internal/runner/command_test.go

```diff
@@ -0,0 +1,132 @@
+package runner
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/kballard/go-shellquote"
+)
+
+func TestRunAndForwardSignal(t *testing.T) {
+	cmd := exec.Command("echo", "hello world")
+
+	err := runAndForwardSignal(cmd)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestRunAndForwardSignal_CommandExitsWithNonZero(t *testing.T) {
+	cmd := exec.Command("false")
+
+	err := runAndForwardSignal(cmd)
+	exitError := new(exec.ExitError)
+	if errors.As(err, &exitError) && exitError.ExitCode() != 1 {
+		t.Errorf("Expected exit code 1, but got %d", exitError.ExitCode())
+	}
+}
+
+func TestRunAndForwardSignal_SignalReceivedInMainProcess(t *testing.T) {
+	cmd := exec.Command("sleep", "10")
+
+	// Send a SIGTERM signal to the main process.
+	go func() {
+		pid := os.Getpid()
+		process, err := os.FindProcess(pid)
+		if err != nil {
+			fmt.Println(err)
+		}
+		time.Sleep(300 * time.Millisecond)
+		process.Signal(syscall.SIGTERM)
+	}()
+
+	err := runAndForwardSignal(cmd)
+
+	signalError := new(ProcessSignaledError)
+	if !errors.As(err, &signalError) {
+		t.Errorf("Expected ErrProcessSignaled, but got %v", err)
```

Comment:
Is `Expected ... got...` a pattern that the rest of the project has fallen into? 
I recommend [got-before-want](https://google.github.io/styleguide/go/decisions#got-before-want), but if this is how you (as a team) prefer to do it then apply consistency and adjust my suggestions elsewhere.

Reply:
Thanks! We use `got-before-want`. I just have this tendency to switch to `Expected ... got ...` When comparing things that are not the output of that function (e.g. error). But your suggestions in other comment makes more sense, and consistent with `got-before-want`.